### DISCUSSION TITLE
Uno How to find objected for named control

### DISCUSSION THREAD

**Discussion by HowardPWeiss** [unoplatform/uno#16862] [Q&A]
XAML as follows

```xaml
<Page
    x:Class="Hello.Presentation.MainPage"
    xmlns="http://schemas.microsoft.com/winfx/2006/xaml/presentation"
    xmlns:x="http://schemas.microsoft.com/winfx/2006/xaml"
    xmlns:local="using:Hello.Presentation"
    xmlns:d="http://schemas.microsoft.com/expression/blend/2008"
    xmlns:mc="http://schemas.openxmlformats.org/markup-compatibility/2006"
   
    mc:Ignorable="d"    
    Background="{ThemeResource ApplicationPageBackgroundThemeBrush}"
    Height="839" Width="1672">

    <Page.DataContext>
        <local:MainViewModel/>
    </Page.DataContext>
    <Page.Resources>

 
    </Page.Resources>

    <Grid Height="839" Width="1672" Background="Black">
        <Button Name="btnReset" Command="{Binding btnReset_Click}" IsEnabled="{Binding Path=bResetEnabled}" Height="27" Width="96" TabIndex="4" Margin="13,20,0,0" HorizontalAlignment="Left" 		VerticalAlignment="Top" >Reset</Button>

    </Grid>
</Page>
```

In my ViewModel, I would like to find the object corresponding to the button btnReset

Is there an equivalent of the WPF method FrameworkElement.FindName(String) ?

I have been able to walk the object tree by doing the following

Modify Main Page class to remember the MainPage Instance

```csharp
public sealed partial class MainPage : Page
{
	public static MainPage MainPageInstance;

        public MainPage()
        {
             MainPageInstance = this;
             this.InitializeComponent();
        }
}
```

I can then walk the object tree using code such as
```csharp
Hello.Presentation.MainPage mp = MainPage.MainPageInstance as Hello.Presentation.MainPage;
Microsoft.UI.Xaml.Controls.Grid gr = mp.Content as Microsoft.UI.Xaml.Controls.Grid;
Microsoft.UI.Xaml.Controls.ItemsControl ic = gr.Children[44] as Microsoft.UI.Xaml.Controls.ItemsControl;
Microsoft.UI.Xaml.Controls.ItemCollection ix = ic.Items as Microsoft.UI.Xaml.Controls.ItemCollection;
Microsoft.UI.Xaml.Controls.Canvas cv = ic.ItemsPanelRoot as Microsoft.UI.Xaml.Controls.Canvas;
Microsoft.UI.Xaml.Controls.UIElementCollection co = cv.Children as Microsoft.UI.Xaml.Controls.UIElementCollection;
```

which I find abhorrent

**Comment by jeromelaban** [comment]
@HowardPWeiss thanks for the question. Yes, you can use [`FindName`](https://learn.microsoft.com/en-us/uwp/api/windows.ui.xaml.frameworkelement.findname?view=winrt-22621) to find controls in the same way you could, though you need to use [`x:Name`](https://learn.microsoft.com/en-us/windows/uwp/xaml-platform/x-name-attribute) instead.

**Reply by jeromelaban** [reply]
@HowardPWeiss you can find the usage of [x:Name at this location](https://learn.microsoft.com/en-us/windows/uwp/xaml-platform/x-name-attribute#xaml-attribute-usage).

**Reply by jeromelaban** [reply]
_Please make sure to respond the same thread, it makes it easier for the community to follow the discussion_

>  I believe I used x:Name correctly in my XAML. My question was how to invoke FindElement since it requires an object reference and I am not sure how I obtain the needed object reference

@HowardPWeiss You did not use it correctly, the syntax for [`x:Name`](https://learn.microsoft.com/en-us/windows/uwp/xaml-platform/x-name-attribute) is as follows:

```xaml
<Button x:Name="MyButton" />
```

Which can be used with the instance function `FindName`, like `this.FindName("MyButton");` in your code behind class.

**Comment by HowardPWeiss** [comment]
Jerome

Thank you for your response.  Please clarify how to use FindElement

My .xaml file lools like

<Page
  <Grid
      <Canvase Name="X:Gameboard"

The documentation for FindElement says it is a method of the FrameWork.Element

In my MainViewModell constructor, I tried

   FrameworkElement.FindName("Gameboard");

This does not compile

Error: error CS0120: An object reference is required for the non-static field, method, or property 'FrameworkElement.FindName(string)' 

What object reference do I use?

**Comment by HowardPWeiss** [comment]
Jerome -

I believe I used x:Name correctly in my XAML.

My question was how to invoke FindElement since it requires an object reference and I am not sure how I obtain the needed object reference
